### ISSUE TITLE
aimet_torch.onnx.export emits non-canonical QDQ pattern for nn.Linear weights with Transpose after DequantizeLinear

### ISSUE THREAD

**Issue by yyh-603** [quic/aimet#4105]
## Description

I found that `aimet_torch.onnx.export` exports quantized `nn.Linear` weights with a `Transpose` node placed after `DequantizeLinear`.

For a minimal quantized `nn.Linear` model, the exported ONNX graph contains the following pattern:

```text
QuantizeLinear(fc.weight, axis=0)
  -> DequantizeLinear(axis=0)
  -> Transpose(perm=[1, 0])
  -> MatMul
  -> Add
```

This graph is ONNX-valid. PyTorch `nn.Linear` stores weights as `[out_features, in_features]` and computes:

```text
x @ weight.T + bias
```

However, this QDQ placement appears to be non-canonical for downstream QNN/HTP compilation. Since the constant weight transpose is placed after `DequantizeLinear`, QNN/HTP does not appear to recognize the layer as a quantized FullyConnected / Linear pattern. As a result, the layer is compiled and executed as floating point after QAIRT compilation.

## Expected behavior

The exported QDQ ONNX graph should use a backend-friendly canonical form for quantized `nn.Linear` weights, for example:

```text
QuantizeLinear(weight.T, axis=1)
  -> DequantizeLinear(axis=1)
  -> MatMul
  -> Add
```

or another equivalent representation that can be recognized as a quantized FullyConnected / Linear operation by QNN/HTP.

In other words, because the weight tensor is a constant initializer, the transpose could be folded into the initializer during export, and the per-channel Q/DQ axis should be updated accordingly.

<img width="1546" height="1501" alt="Image" src="https://github.com/user-attachments/assets/3d058d58-24fe-4a3d-8970-2374e175f63b" />

## Actual behavior

`aimet_torch.onnx.export` emits:

```text
QuantizeLinear(fc.weight, axis=0)
  -> DequantizeLinear(axis=0)
  -> Transpose(perm=[1, 0])
  -> MatMul
  -> Add
```

This keeps the constant weight transpose as a graph operation after `DequantizeLinear`. In my QAIRT/QAI Hub compilation flow, this causes the Linear/FC layer to fall back to floating-point execution instead of being compiled as a quantized HTP FullyConnected / Linear operation.

<img width="2870" height="2998" alt="Image" src="https://github.com/user-attachments/assets/aa80af58-886f-42e0-9a87-c14a161ecf0a" />

## Minimal reproduction

```python
import torch
import torch.nn as nn

from aimet_torch import QuantizationSimModel
from aimet_torch.model_preparer import prepare_model
from aimet_torch.common.defs import QuantScheme
import aimet_torch


DEVICE = "cuda" if torch.cuda.is_available() else "cpu"


class TinyLinearModel(nn.Module):
    def __init__(self):
        super().__init__()
        self.fc = nn.Linear(1280, 1280, bias=True)

    def forward(self, x):
        return self.fc(x)


@torch.no_grad()
def calibrate(model):
    model.eval()
    for _ in range(8):
        x = torch.randn(1, 257, 1280, device=DEVICE)
        _ = model(x)


def main():
    print(f"Using device: {DEVICE}")

    model = TinyLinearModel().to(DEVICE).eval()
    dummy_input = torch.randn(1, 257, 1280, device=DEVICE)

    model = prepare_model(model)
    model.to(DEVICE).eval()

    sim = QuantizationSimModel(
        model,
        dummy_input,
        default_output_bw=8,
        default_param_bw=8,
        in_place=True,
        config_file="htp_v69",
        quant_scheme=QuantScheme.post_training_tf_enhanced,
    )

    sim.compute_encodings(calibrate)

    onnx_path = "tiny_linear_aimet.onnx"

    aimet_torch.onnx.export(
        sim,
        dummy_input,
        f=onnx_path,
        input_names=["input"],
        output_names=["output"],
        opset_version=21,
        dynamo=True,
        export_int32_bias=True,
        external_data=False,
    )


if __name__ == "__main__":
    main()
```

After generating the ONNX model, I compiled it through QAI Hub / QAIRT for the target device.

The model is exported through the following AIMET flow:

```text
prepare_model
  -> QuantizationSimModel
  -> compute_encodings
  -> aimet_torch.onnx.export
```

## Configuration

```text
Python version: 3.10
aimet-torch version: 2.32.1
PyTorch version: 2.12.0
ONNX version: 1.21.0
QAIRT version: 2.45.0.260326154327
AI Hub Workbench version: aihub-2026.05.27.0
Target backend: HTP v69
Target device: XR2 Gen 2 (Proxy)
```

I also tested other backend versions, including v73, v75, and v79, and observed the same fallback behavior.

Please let me know if this is expected behavior, or if there is a recommended export configuration to canonicalize constant Linear weights before QDQ insertion.

**Comment by kyunggeunlee** [comment]
This is a very good point. There has been an internal request/discussion that QAIRT should support `W-QDQ-Transpose-MatMul` pattern seamlessly without floating point fallback, but there hasn't been much progress yet.
But your suggestion sounds good to me regardless. Let us work on it

**Comment by kyunggeunlee** [comment]
We've created an internal AIMET PR to work around this issue, and it's expected to be shipped in the next aimet release.
Independently, QAIRT team has also started work to support W-QDQ-Transpose-MatMul pattern.

In the meantime, you can try using [**sim**.onnx.export](https://quic.github.io/aimet-pages/releases/latest/apiref/torch/quantsim.html#aimet_torch.QuantizationSimModel.onnx) instead to get yourself unblocked.
This is basically the same as [**aimet_torch**.onnx.export](https://quic.github.io/aimet-pages/releases/latest/apiref/torch/onnx.html#apiref-torch-onnx), but it exports the encodings in a separate json file (encoding in ".encodings").
We observed that QAIRT can handle `W-QDQ-Transpose-MatMul` when it's exported as json encoding, but not when it's exported as onnx QDQ.

**Comment by kyunggeunlee** [comment]
Resolved by a203e14. This fix will be shipped in aimet 2.34 on Jun 29
